documentation/builders/event-devices.md

```diff
@@ -0,0 +1,134 @@
+        device_name: {device_name} # name of the evdev
+        exact: False/True # optional to require exact match
+        mapping:
+          {key-code}: # evdev event id
+              {rpc_command_definition} # eg `alias: toggle`
+```
+The `{device nickname}` is only for your own orientation and can be choosen freely.
+For each device you need to figure out the `{device_name}` and the `{event_id}` corresponding to key strokes, as indicated in the sections below.
+
+### Identifying the `{device_name}`
+
+The `{device_name}` can be identified using the following Python snippet:
+
+``` Python
+  import evdev
+  devices = [evdev.InputDevice(path) for path in evdev.list_devices()]
+  for device in devices:
+      print(device.path, device.name, device.phys)
+```
+
+The output could be in the style of:
+
+```
+  /dev/input/event1    Dell Dell USB Keyboard   usb-0000:00:12.1-2/input0
+  /dev/input/event0    Dell USB Optical Mouse   usb-0000:00:12.0-2/input0
+```
+
+In this example, the `{device_name}` could be `DELL USB Optical Mouse`.
+Note that if you use the option `exact: False`, it would be sufficient to add a substring such as `USB Keyboard`.
+
+### Identifying the `{key-code}`
+
+The key code for a button press can be determined using the following code snippet:
+
+``` Python
+  import evdev
+  device = evdev.InputDevice('/dev/input/event0')
+  device.capabilities(verbose=True)[('EV_KEY', evdev.ecodes.EV_KEY)]
+```
+
+With the `InputDevice` corresponding to the path from the output of the section `{device_name}` (eg. in the example `/dev/input/event0`
+would correspond to `Dell Dell USB Keyboard`).
+
+If the naming is not clear, it is also possible to empirically check for the key code by listening for events:
+
+``` Python
+  from evdev import InputDevice, categorize, ecodes
+  dev = InputDevice('/dev/input/event1')
+  print(dev)
+  for event in dev.read_loop():
+    if event.type == ecodes.EV_KEY:
+      print(categorize(event))
+```
+The output could be of the form:
+```
+  device /dev/input/event1, name "DragonRise Inc.   Generic   USB  Joystick  ", phys "usb-3f980000.usb-1.2/input0"
+  key event at 1672569673.124168, 297 (BTN_BASE4), down
+  key event at 1672569673.385170, 297 (BTN_BASE4), up
+```
+
+In this example output, the `{key-code}` would be `297`
+
+Alternatively, the device could also be setup without a mapping.
+Afterwards, when pressing keys, the key codes can be found in the log files. Press various buttons on your device,
+while watching the logs with `tail -f shared/logs/app.log`.
+Look for entries like `No callback registered for button ...`.
+
+### Specifying the `{rpc_command_definition}`
+
+Each key looks like `{key-code}: {rpc_command_definition}`.
+The RPC command follows the regular RPC command rules as defined in the [following documentation](./rpc-commands.md).
+
+
+## Full example config
+
+Here is a complete configuration example for a USB Joystick controller:
```

Comment:
I wonder, how close can we get this definition to GPIO? It'll be important to maintain parity of functionality between the two input types to reduce confusion with users.

I am referring to 1) the yaml layout (like for [GPIO](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/resources/default-settings/gpio.example.yaml)) and 2) the actions types such es "press", "short/long press", rotary actions etc.

Reply:
Aiming to harmonize the interface with GPIO seems valuable.

However I doubt that a 1:1 mapping of the config is feasible as the GPIO interface supports a single device with multiple pins, while EVDEV would be multiple devices with multiple pins.

`short/long press` does need additional logic (see discussion here: https://github.com/gvalkov/python-evdev/issues/70).
Things like rotation may be feasible as well, but would need more investigation. In particular it would be helpful to have a device at hand that has such events as a concrete use case.
Output (toggling LEDs, sounds) are also supported by EVDEV, but I dont have a device at hand to test it.

My proposal forward would be that I try to restructure the yaml that it follows the `gpio` interface more closely and maybe add stubs for `short/long press` with an `NotImplementedError`.
Once merged we may add follow up issues to investigate the future directions.

Reply:
Sounds like a good idea. If a full harmonization is not possible, I agree it's still valuable to get as close as possible and document the differences.